product_by_supplier/views/product_supplierinfo_view.xml

```diff
@@ -56,11 +46,10 @@
             <field name="inherit_id" ref="product.product_supplierinfo_form_view"/>
             <field name="priority">99</field>
             <field name="arch" type="xml">
-                <xpath expr="//field[@name='product_name']" position="before">
-                    <field name="product_tmpl_id" string="Product"
-                    invisible="context.get('from_product_form')"
-                    required="not context.get('from_product_form')"/>
-                </xpath>
+                <field name="product_tmpl_id" position="replace"/>
+                <field name="product_id" position="before">
+                    <field name="product_tmpl_id" required="True" invisible="context.get('default_product_tmpl_id')"/>
+                </field>
```

Comment:
@ivantodorovich 
I think that the field should not be replaced instead you could use a position="attributes" to add the new characteristic you need.

Reply:
Hi @zaoral !

That was my initial intention. But the original location of the field on the view is not as good.
In fact, as in Odoo it's never shown, I think they never wasted a second thinking on the location, they've just placed it there.

Reply:
@ivantodorovich 

mmm.. ok I get it. The problem is that you want to move the location of the field.

First, I think that replace a field completly from a view specially if this one came from odoo addons directly is a delicate thing to do: I understand your claim but I think we should not replace the field IMHO. 

Sadly as far as I know we can't move a xml node (field) to another place in the view: if that is possible well would be perfect.

However, in this case, I have to agree with @pedrobaeza. The best option is create a new view (inherit from the original but explicitly set as primary in order to do not overwrite, only copy the complete view) will be the best approach, there you can replace the field and add what you like.